java/ddlogapi/DDlogAPI.java

```diff
@@ -659,33 +659,85 @@ public static String ddlogInstallationPath() throws DDlogException {
         return ddlogHome;
     }
 
+    public static class CompilationResult {
```

Comment:
Do some of the methods in this class need to be synchronized because they can be invoked by the client as well as by the output reader threads this class spawns?